common/app/services/newsletters/NewsletterSignupLifecycle.scala

```diff
@@ -0,0 +1,40 @@
+package services.newsletters
+
+import app.LifecycleComponent
+import common.JobScheduler
+import play.api.inject.ApplicationLifecycle
+
+import java.util.concurrent.Executors
+import scala.concurrent.{ExecutionContext, ExecutionContextExecutorService, Future}
+
+class NewsletterSignupLifecycle(
+    appLifecycle: ApplicationLifecycle,
+    jobs: JobScheduler,
+    newsletterSignupAgent: NewsletterSignupAgent,
+)(implicit
+    ec: ExecutionContext,
+) extends LifecycleComponent {
+
+  appLifecycle.addStopHook { () =>
+    Future {
+      descheduleAll()
+    }
+  }
+
+  private def descheduleAll(): Unit = {
+    jobs.deschedule("NewsletterSignupAgentLowFrequencyRefreshJob")
+  }
+
+  override def start(): Unit = {
+
+    descheduleAll()
+    newsletterSignupAgent.refreshNewsletters()
+    newsletterSignupAgent.refreshGroupedNewsletters()
```

Comment:
I would add a method `refresh` to the agent that does both of these individual refreshes. It exposes less of the internal logic of the agent outside.

Reply:
What would you use for the return type on it? 
`refreshNewsletters()` gives
```scala
Future[Either[String, List[NewsletterResponse]]]
```
whilst `refreshGroupedNewsletters()` gives
```scala
Future[Either[String, GroupedNewslettersResponse]]
```

I wouldn't want to squash their errors down to a single field because it could be that one end point could fail whilst another didn't.
I could do it as a list I guess but it would change from from being `Future`s